src/states/elder/mod.rs

```diff
@@ -424,7 +421,27 @@ impl Elder {
 
         let our_pfx = *self.our_prefix();
 
-        let (to_vote_again, to_process) = cached_events
+        let to_process = cached_events
+            .iter()
+            .filter(|event| match &event.payload {
+                // Events to re-process
+                AccumulatingEvent::Online(_) => !completed_events.contains(&event.payload),
```

Comment:
Minor: As all non-completed `Online` go to `to_process`, you can just filter them out unconditionally in the `to_vote_again` collection below.

Reply:
Not quite, we expect them to be re-voted too in the current scheme.

Reply:
Ah true. OK.